core/src/main/java/org/opensearch/sql/expression/datetime/IntervalClause.java

```diff
@@ -93,7 +93,7 @@ private ExprValue hour(ExprValue value) {
   }
 
   private ExprValue day(ExprValue value) {
-    return new ExprIntervalValue(Duration.ofDays(getIntegerValue(value)));
+    return new ExprIntervalValue(Period.ofDays(getIntegerValue(value)));
```

Comment:
Why change from Duration to Period?

Reply:
It was required by changes in a4bca4f9:
https://github.com/Bit-Quill/opensearch-project-sql/blob/a4bca4f978984e1b2097a9c1d303521edfffb6a5/core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java#L467-L483
These changes were mostly reverted, because they added ambiguous function signatures:
https://github.com/Bit-Quill/opensearch-project-sql/blob/a4bca4f978984e1b2097a9c1d303521edfffb6a5/core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java#L109-L110
> `returnType, arg1Type, arg2Type`

I still want to keep this change regardless of revert, because this make `INTERVAL` more consistent:
https://github.com/Bit-Quill/opensearch-project-sql/blob/2c1e83d1e1144b8d488e974fa48f2e7dc71d8b03/core/src/main/java/org/opensearch/sql/expression/datetime/IntervalClause.java#L87-L101

`Period` for `Date`-like intervals and `Duration` for `Time`-like intervals